packages/jest-cli/src/cli/processArgs.js

```diff
@@ -221,8 +223,26 @@ function processArgs() {
       if (argv.testEnvData) {
         argv.testEnvData = JSON.parse(argv.testEnvData);
       }
+
+      const yargsSpecialOptions = ['$0', '_'];
+      const allowedOptions = Object.keys(options).reduce((acc, option) => {
+        return acc
+          .add(option)
+          .add(options[option].alias);
+      }, new Set(yargsSpecialOptions));
+      const unrecognizedOptions = Object.keys(argv).filter(arg => {
+        return !allowedOptions.has(arg);
+      });
+      if (unrecognizedOptions.length) {
+        throw new Error(
```

Comment:
The trouble is the "silent accept" failure mode is really nasty in the case where you mean to filter to a subset of the test files.

```
# Without the throw, this might run 1,000 tests:
jest --fast MyModule
```

But I guess a warning would be ok. Although in the scenario above, I might not even see the warning among the output from the 1,000 files. Your call!